app/controllers/users/email_confirmations_controller.rb

```diff
@@ -44,10 +44,14 @@ def process_successful_confirmation(email_address)
       confirm_and_notify(email_address)
       if current_user
         flash[:success] = t('devise.confirmations.confirmed')
-        redirect_to account_url
+        if params[:request_id]
+          redirect_to sign_up_select_email_url
+        else
+          redirect_to account_url
```

Comment:
I wondered if we could use [`after_sign_in_path_for`](https://github.com/18F/identity-idp/blob/bed2a55da0f156d7226f1e871f7c9ee4dcbe54ac/app/controllers/application_controller.rb#L226-L246) to figure out where they should go, which already handles `request_id` to send to consent screen or account page. I guess the risk is there's a lot of additional checks there that I'm not entirely sure we'd want to consider as part of confirming an email address.

Reply:
I experimented some with that but it wasn't ending up where we wanted to be. I agree there are a lot of checks in there that aren't needed I think.

Reply:
> I experimented some with that but it wasn't ending up where we wanted to be.

Where was it going? I'm wondering if this could be related to [my previous comment](https://github.com/18F/identity-idp/pull/11439#discussion_r1825703769) about using `StoreSpMetadataInSession` to ensure that the session is initialized with the `request_id`.

Reply:
reply ended up here https://github.com/18F/identity-idp/pull/11439#discussion_r1827955453